Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyQuestions.kt

```diff
@@ -19,216 +19,74 @@ package com.example.compose.jetsurvey.survey
     ) {
-        item {
-            Spacer(modifier = Modifier.height(32.dp))
-            QuestionTitle(question.questionText)
-            Spacer(modifier = Modifier.height(24.dp))
-            if (question.description != null) {
-                Text(
-                    text = stringResource(id = question.description),
-                    color = MaterialTheme.colorScheme.onSurface
-                        .copy(alpha = stronglyDeemphasizedAlpha),
-                    style = MaterialTheme.typography.bodySmall,
-                    modifier = Modifier
-                        .fillParentMaxWidth()
-                        .padding(bottom = 18.dp, start = 8.dp, end = 8.dp)
-                )
-            }
-            when (question.answer) {
-                is PossibleAnswer.SingleChoice -> SingleChoiceQuestion(
-                    options = question.answer.options,
-                    answer = answer as Answer.SingleChoice?,
-                    onAnswerSelected = { answer -> onAnswer(Answer.SingleChoice(answer)) },
-                    modifier = Modifier.fillMaxWidth()
-                )
-                is PossibleAnswer.MultipleChoice -> MultipleChoiceQuestion(
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.MultipleChoice?,
-                    onAnswerSelected = { newAnswer, selected ->
-                        // create the answer if it doesn't exist or
-                        // update it based on the user's selection
-                        if (answer == null) {
-                            onAnswer(Answer.MultipleChoice(setOf(newAnswer)))
-                        } else {
-                            onAnswer(answer.withAnswerSelected(newAnswer, selected))
-                        }
-                    },
-                    modifier = Modifier.fillMaxWidth()
-                )
-                is PossibleAnswer.Action -> ActionQuestion(
-                    questionId = question.id,
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.Action?,
-                    onAction = onAction,
-                    modifier = Modifier.fillParentMaxWidth()
-                )
-                is PossibleAnswer.Slider -> SliderQuestion(
-                    possibleAnswer = question.answer,
-                    answer = answer as Answer.Slider?,
-                    onAnswerSelected = { onAnswer(Answer.Slider(it)) },
-                    modifier = Modifier.fillParentMaxWidth()
-                )
-            }
+        Spacer(Modifier.height(32.dp))
+        QuestionTitle(titleResourceId)
+        directionResourceId?.let {
+            Spacer(Modifier.height(18.dp))
+            QuestionDirections(it)
         }
+        Spacer(Modifier.height(18.dp))
+
+        content()
     }
 }
 
 @Composable
-private fun QuestionTitle(@StringRes title: Int) {
-    Row(
+fun QuestionTitle(@StringRes title: Int) {
```

Comment:
Make composables private that are only extracted for readability and not reusability. Goes for the whole project.

Reply:
Updated `QuestionTitle` and `QuestionDirections` to be private.